pkg/extensions/sync/httpclient/client.go

```diff
@@ -298,7 +284,15 @@ func (httpClient *Client) makeAndDoRequest(method, mediaType, namespace, urlStr
 		return nil, nil, err
 	}
 
-	if err := httpClient.setupAuth(req, namespace); err != nil {
+	err = httpClient.setupAuth(req, namespace)
+	if err != nil {
+		// harbor catalog requests return basicAuth by default, even if bearer is used on the rest of endpoints.
+		if errors.Is(err, zerr.ErrUnexpectedAuthHeader) &&
```

Comment:
Based on https://github.com/goharbor/harbor/blob/9e8e647b713b9e33d9b8c32adbad066f1d7b5bf6/src/server/middleware/v2auth/auth.go#L99 they expect to receive an auth header for CLIs.

Maybe instead of this weird workaround we should send an auth header when we make the /v2/_catalog request? Don't we already know what headers to use from the initial /v2/ call?

Reply:
Actually it's the other way around, if they receive and auth header on  /v2/_catalog, they return basic auth. Why would they the do this?